sos/plugins/sas3ircu.py

```diff
@@ -0,0 +1,47 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+
+# This sosreport plugin is meant for sas adapters.
+# This plugin logs inforamtion on each adapter it finds.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+import os
+
+
+class SAS3ircu(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+
+    plugin_name = "sas3ircu"
+
+    def setup(self):
+
+        # get list of adapters
+        output = self.call_ext_prog("sh -c 'sas3ircu list'", timeout=5)
+
```

Comment:
Nack. What is the point of the shell here? (generally plugins must not invoke a shell unless there is a very good reason for it - we eliminated the default shell on external processes years ago because it opens us up to a whole class of security problems).

Since this isn't using any shell notation there does not seem any point to it (and if there is then it should be clearly explained in a comment, so that someone reading the code knows what is going on, and can possibly find a better solution).